File: backend/Domain/Entities/Service.cs
namespace Domain.Entities;

public class Service
{
    public Guid Id { get; set; }

    public string Name { get; set; } = null!;

    public string? Description { get; set; }

    public decimal Price { get; set; }

    public DateTime CreatedAt { get; set; }

    public Guid? CreatedBy { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public Guid? UpdatedBy { get; set; }

    public DateTime? DeletedAt { get; set; }

    public Guid? DeletedBy { get; set; }

    public bool IsDeleted { get; set; }

    public ICollection<Feedback> Feedbacks { get; set; } = [];

    public ICollection<Media> Media { get; set; } = [];

    public ICollection<OrderDetail> OrderDetails { get; set; } = [];
}

File: frontend/src/Interfaces/Service/Types/Service.ts
import z from 'zod/v4'

export interface Service {
	id: string
	name: string
	description: string
	price: number
	createdAt: Date
	updatedAt?: Date | null
	isDeleted: boolean
	deletedBy?: string | null
	createdBy?: string | null
	imageUrls?: Array<{ id: string; url: string }>
}

export const updateServiceSchema = z.object({
	name: z.string().min(1, 'Name is required'),
	description: z.string().min(1, 'Description is required'),
	price: z.coerce.number().positive('Price must be greater than 0'),
	isDeleted: z.boolean(),
	deletedBy: z.string().optional().nullable(),
	createdBy: z.string().optional().nullable(),
	imageUrls: z
		.array(
			z.object({
				id: z.string(),
				url: z.url('Each image must be a valid URL'),
			})
		)
		.optional(),
})

export type UpdateServiceApiRequest = z.infer<typeof updateServiceSchema>
